lib/compat/wordpress-6.5/block-bindings/class-wp-block-bindings-registry.php

```diff
@@ -132,13 +158,65 @@ public function register( string $source_name, array $source_properties ) {
 				return false;
 			}
 
+			// Validate that the uses_context parameter is an array.
+			if ( isset( $source_properties['uses_context'] ) && ! is_array( $source_properties['uses_context'] ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					__( 'The "uses_context" parameter must be an array.' ),
+					'6.5.0'
+				);
+				return false;
+			}
+
+			// Validate that the source properties contain only allowed properties.
+			if ( ! empty( array_diff( array_keys( $source_properties ), $this->allowed_source_properties ) ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					__( 'The $source_properties array contains invalid properties.' ),
+					'6.5.0'
+				);
+				return false;
+			}
+
 			$source = new WP_Block_Bindings_Source(
 				$source_name,
 				$source_properties
 			);
 
 			$this->sources[ $source_name ] = $source;
 
+			// Add `uses_context` defined by block bindings sources.
+			// If `get_uses_context` exists (from WP 6.5), use its filter because it is more reliable.
+			if ( method_exists( 'WP_Block_Type', 'get_uses_context' ) ) {
```

Comment:
This is a compat layer for WP 6.5, so this class is going to be used in WP 6.3 and 6.4. In effect, this check is not needed. If I follow the flow correctly, only `register_block_type_args` filter is necessary here because in older version of WP there is no `get_uses_context` method.

Reply:
Maybe include a comment that it differs intentionally from WP core so it doesn't get replaced by accident.

Reply:
I didn't see this one, sorry 🙏 I can do it in a follow-up PR.